### DISCUSSION TITLE
[Hackathon Use Case] - LCR on Morphir

### DISCUSSION THREAD

**Discussion by maoo** [finos/finos-pmcs#245] [Ideas]
# LCR on Morphir
_From @stephengoldbaum (Morgan Stanley)_

## Business Problem

Given the goal of providing the regulation as code, a core requirement is representing the business concepts defined in the specifiation as computer concepts. We do this by implementing the LCR using Morphir by way of the open-source [Elm](https://elm-lang.org/) programming language. In order to be complete, it is important that every business concept is fully represented. This requires interpreting the specification to understand the best approach to implement each concept in Morphir. This document reviews that process and the decisions made.

More info on https://github.com/finos/open-reg-tech-us-lcr/blob/main/doc/README.md

## Expected outcome

We left part of the [LCR](https://github.com/finos/open-reg-tech-us-lcr) unfinished to gain experience with open-source collaboration on a regulation. There are two main items to be finished:

- Finish the unfinished logic - The various parts of the LCR are independent, so we can parcel this out across a number of people.
- Test data - We need a quality set of test data to run under realistic conditions. We can look at synthetic data generation and other options.

## Useful apps, systems, projects, technologies, libraries and datasets

- [Current LCR implementation](https://github.com/finos/open-reg-tech-us-lcr)
- [LCR interactive](http://lcr-interactive.finos.org/)
- [Morphir](https://morphir.finos.org)
- [Data Helix generator](https://finos.github.io/datahelix/)
- [DataHub](https://datahub.finos.org/) - produce synthetic datasets
- [synthesized.io](https://www.synthesized.io/) - Create high-quality data for machine learning, application development and testing with easy-to-use config files

**Comment by robmoffat** [comment]
"Help Build Open Source Regulation"

**Comment by smorres** [comment]
Hi @stephengoldbaum,

Our team spoke with @maoo today about potentially working on this use case for the upcoming hackathon. Could we set up a call with you to get your expert insight on the project and better understand the code? Please let me know if this is possible. Thank you! -Shane

**Reply by stephengoldbaum** [reply]
Hi Shane,

That's great to hear.  We can definitely review. Let's setup some time this or next week.  

Stephen

**Reply by smorres** [reply]
Awesome- what is the best way to contact you directly?

**Reply by mcleo-d** [reply]
Hi @smorres - Did this thread get resolved as I noticed GitHub filtered @stephengoldbaum's email address?

**Reply by maoo** [reply]
yes, all sorted, thank @mcleo-d !

**Comment by stephengoldbaum** [comment]
***@***.***

**Comment by maoo** [comment]
Here [a setup guide](https://github.com/finos/open-reg-tech-us-lcr/blob/main/doc/contribute.md) to help prepare the development environment.  We’re going to pitch 4 considerations for the group, which we will vote on.  They are:

1. Finish the regulation rules
2. Create a test harness
3. Improve the transparency user experience
4. Explore ChatGPT for generating portions of the rules implementation.

**Comment by TheJuanAndOnly99** [comment]
Closing discussion in preparation [for this years hackathon](https://www.finos.org/hosted-events/2024-10-25-citi-india-hackathon-supported-by-finos). Feel free to reopen and update if its still relevant.
